ENH: Improve macOS build script

nevack opened the pull request:
#### Changes made here:
 - Update shebang to `/bin/bash`, as it is already expected to be present in `run_gettext.sh`
 - Added fail-fast shell options.
 - Fixed building for non-multi-config generators (Ninja, Make)
 - Added new options:
   - Allow to set CMake generator to Ninja, no option for Makefiles thou.
   - Allow to build without reconfiguring CMake, improves build times. Unnecessary Cmake reconfigurations require full rebuild.
   - Allow to set build configuration for CMake.
 - Reorganized targets into separate function to break order-dependent `cd`s.
 - Reformat shell code.
 - Fix all warnings reported by ShellCheck linter tool.

#### To discuss:
- [x] ~~Should we change by default to Ninja? It's faster, but maybe it should an option for now.~~ Decided to leave Xcode as default generator, but use Ninja on CI with explicit option.
- [x] ~~Should I revert `--full` option unpacking in `run_gettext.sh`? This option is mainly used by the maintainer aka @SoftFever, but we can make it more useful with build output dirs other than hard coded `build_arm64`.~~ Reverted, only linter fixes left in this file.
- [x] ~~Would it be better to split this in several commits or even PRs?~~ Maintainer is ok with this PR as it is.

#### Some other improvements to be made:
1. Update building instructions to be more clear and specify how to build with either Xcode or Ninja
2. Add support for generating `compile_commands.json` with supported CMake generators: Ninja and Makefiles. This will allow to use language completion provided by `clangd` in any supported IDE or editor, like VSCode.
3. Leverage `-S` option of CMake to use less `cd`s.

##### I found it really hard to build this project on macOS, TBH. This can be improved :)

nevack commented:
@SoftFever Could you approve workflows for this PR if the changes seem legit for you?

SoftFever commented:
Thanks!
Gere are some quick feedbacks
> #### To discuss:
> * [ ]  Should we change by default to Ninja? It's faster, but maybe it should an option for now.

I personally use Ninja because of clangd support. However, for the sake of simplicity and wider adoption, I'd like to stick with the default Xcode generator here. Ninja as an extra option will be perfect though.

> * [ ]  Should I revert `--full` option unpacking in `run_gettext.sh`? This option is mainly used by the maintainer aka @SoftFever, but we can make it more useful with build output dirs other than hard coded `build_arm64`.

Checking the computer's architecture to set up the folders is a better way, but we also need to prepare hintsToPot first. I'm not sure if anyone else uses this method, though.

> * [ ]  Would it be better to split this in several commits or even PRs?

I'm fine with one single PR as they are related

SoftFever commented:
> @SoftFever Could you approve workflows for this PR if the changes seem legit for you?

I like these changes 👍 
We just need to make sure the CI/CD pipeline works

nevack commented:
> I personally use Ninja because of clangd support. However, for the sake of simplicity and wider adoption, I'd like to stick with the default Xcode generator here. Ninja as an extra option will be perfect though.

Same here. I also want to update building instructions with the info how to use this setup next.
I'm fine with Xcode by default and an option for Ninja.
Should I enable Ninja on CI with this new explicit option?

> Checking the computer's architecture to set up the folders is a better way, but we also need to prepare hintsToPot first. I'm not sure if anyone else uses this mode, though. So I'm not bothered.

Yep. I think I will revert changes in `run_gettext.sh` for now, as I don't really like this not so good parsing of options.
You are the only one using it, and your scenario is broken as there're new mandatory options to be provided.

SoftFever commented:
> Should I enable Ninja on CI with this new explicit option?

Sounds good

> Yep. I think I will revert changes in `run_gettext.sh` for now, as I don't really like this not so good parsing of options. You are the only one using it, and your scenario is broken as there're new mandatory options to be provided.

👍

nevack commented:
> I like these changes 👍
> We just need to make sure the CI/CD pipeline works

Thanks.
I think we need to force trigger `Build Deps` workflow family here, right now they are skipped.
Is there simple way to do that?

SoftFever commented:
If we change something in the deps folder, it will trigger build_deps action.
Try add a comment in deps\CMakeLists.txt, it should do the trick

nevack commented:
> If we change something in the deps folder, it will trigger build_deps action. Try add a comment in deps\CMakeLists.txt, it should do the trick

Aha, added a line there.

Also updated everything accordingly to what we discussed, but once again I need an approval for workflows after.

SoftFever commented:
It's running now.
Currently approval is needed to run actions for first-time contributors based on the rule in case you wondered

nevack commented:
> It's running now. Currently approval is needed to run actions for first-time contributors based on the rule in case you wondered

Yeah, I understand.

Builds seems fine https://github.com/SoftFever/OrcaSlicer/actions/runs/8127269674?pr=4310

Rebased to latest main and removed the last commit that was for force rebuilding deps.

SoftFever commented:
> @SoftFever Is all good with CI pipeline? I think it's ready to merge if so :)

The CI builds fine.
I haven't got chance to test the build on Mac though.
Could you help download the build and give it a try on your Mac?

nevack commented:
> Could you help download the build and give it a try on your Mac?

Yep, already did.
By saying "Build seems fine" I meant that I tested artifact from Action run.
Sorry, I needed to be more explicit about that.

SoftFever commented:
> > Could you help download the build and give it a try on your Mac?
> 
> Yep, already did. By saying "Build seems fine" I meant that I tested artifact from Action run. Sorry, I needed to be more explicit about that.

Awesome!
Thank you very much.
Good to merge

igiannakas commented:
Hi all

seems like the pr broke the script when running on arm64 local build. Anyone else encountered this?

superfloh247 commented:
yes, broken on MacOS 14.3.1 arm64

(base)lindner@M1pro OrcaSlicer % ./build_release_macos.sh 
Build params:
 - ARCH: arm64
 - BUILD_CONFIG: Release
 - BUILD_TARGET: all
 - CMAKE_GENERATOR: Xcode
 - OSX_DEPLOYMENT_TARGET: 11.3

Building deps...
+ mkdir -p /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/OrcaSlicer_dep_arm64
+ cd /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64
+ '[' 1. '!=' . ']'
+ cmake .. -DDESTDIR=/Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/OrcaSlicer_dep_arm64 -DOPENSSL_ARCH=darwin64-arm64-cc -DCMAKE_BUILD_TYPE=Release -DCMAKE_OSX_ARCHITECTURES:STRING=arm64 -DCMAKE_OSX_DEPLOYMENT_TARGET=11.3
CMake Deprecation Warning at CMakeLists.txt:23 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- CMAKE_OSX_DEPLOYMENT_TARGET: 11.3
-- prusaslicer_add_cmake_project for Apple
-- OrcaSlicer deps DESTDIR: /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/OrcaSlicer_dep_arm64
-- OrcaSlicer dowload dir for source packages: /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/../DL_CACHE
-- OrcaSlicer deps debug build: OFF
OS X SDK Path: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk
OS X Deployment Target: 11.3
CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  PNG/PNG.cmake:29 (orcaslicer_add_cmake_project)
  CMakeLists.txt:213 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  EXPAT/EXPAT.cmake:1 (orcaslicer_add_cmake_project)
  CMakeLists.txt:218 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  Boost/Boost.cmake:13 (orcaslicer_add_cmake_project)
  CMakeLists.txt:223 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  Cereal/Cereal.cmake:5 (orcaslicer_add_cmake_project)
  CMakeLists.txt:226 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  Qhull/Qhull.cmake:2 (orcaslicer_add_cmake_project)
  CMakeLists.txt:227 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  GLEW/GLEW.cmake:5 (orcaslicer_add_cmake_project)
  CMakeLists.txt:228 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  GLFW/GLFW.cmake:15 (orcaslicer_add_cmake_project)
  CMakeLists.txt:229 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  OpenCSG/OpenCSG.cmake:2 (orcaslicer_add_cmake_project)
  CMakeLists.txt:230 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  TBB/TBB.cmake:1 (orcaslicer_add_cmake_project)
  CMakeLists.txt:232 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  Blosc/Blosc.cmake:29 (orcaslicer_add_cmake_project)
  CMakeLists.txt:234 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  OpenEXR/OpenEXR.cmake:39 (orcaslicer_add_cmake_project)
  CMakeLists.txt:235 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  OpenVDB/OpenVDB.cmake:9 (orcaslicer_add_cmake_project)
  CMakeLists.txt:236 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  GMP/GMP.cmake:59 (ExternalProject_Add)
  CMakeLists.txt:238 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  MPFR/MPFR.cmake:28 (ExternalProject_Add)
  CMakeLists.txt:239 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  CGAL/CGAL.cmake:1 (orcaslicer_add_cmake_project)
  CMakeLists.txt:240 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  NLopt/NLopt.cmake:1 (orcaslicer_add_cmake_project)
  CMakeLists.txt:242 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  OpenSSL/OpenSSL.cmake:41 (ExternalProject_Add)
  CMakeLists.txt:244 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  CURL/CURL.cmake:59 (orcaslicer_add_cmake_project)
  CMakeLists.txt:248 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  JPEG/JPEG.cmake:14 (orcaslicer_add_cmake_project)
  CMakeLists.txt:252 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Compiling TIFF for macos 23.3.0.
CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  TIFF/TIFF.cmake:5 (orcaslicer_add_cmake_project)
  CMakeLists.txt:253 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  wxWidgets/wxWidgets.cmake:29 (orcaslicer_add_cmake_project)
  CMakeLists.txt:254 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  OCCT/OCCT.cmake:11 (orcaslicer_add_cmake_project)
  CMakeLists.txt:255 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.28.3/share/cmake/Modules/ExternalProject.cmake:4171 (message):
  Policy CMP0114 is not set to NEW.  In order to support the Xcode "new build
  system", this project must be updated to set policy CMP0114 to NEW.

  Since CMake is generating for the Xcode "new build system",
  ExternalProject_Add will use policy CMP0114's NEW behavior anyway, but the
  generated build system may not match what the project intends.
Call Stack (most recent call first):
  CMakeLists.txt:120 (ExternalProject_Add)
  FREETYPE/FREETYPE.cmake:13 (orcaslicer_add_cmake_project)
  CMakeLists.txt:256 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring done (0.3s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64
+ cmake --build . --config Release --target deps
Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -project OrcaSlicer-deps.xcodeproj build -target deps -parallelizeTargets -configuration Release -hideShellScriptEnvironment

User defaults from command line:
    HideShellScriptEnvironment = YES
    IDEPackageSupportUseBuiltinSCM = YES

2024-03-06 19:59:44.856 xcodebuild[52092:21716495]  DVTErrorPresenter: Unable to load simulator devices.
Domain: DVTCoreSimulatorAdditionsErrorDomain
Code: 3
Failure Reason: The version of the CoreSimulator framework installed on this Mac is out-of-date and not supported by this version of Xcode.
Recovery Suggestion: Please ensure that you have installed all available updates to your Mac's software, and that you are running the most recent version of Xcode supported by macOS.
--
CoreSimulator is out of date. Current version (944.4.0) is older than build version (947.13.0).
Domain: DVTCoreSimulatorAdditionsErrorDomain
Code: 3
--
2024-03-06 19:59:44.856 xcodebuild[52092:21716495]  iOSSimulator: [SimServiceContext sharedServiceContextForDeveloperDir:error:] returned nil (Error Domain=DVTCoreSimulatorAdditionsErrorDomain Code=3 "CoreSimulator is out of date. Current version (944.4.0) is older than build version (947.13.0)." UserInfo={NSLocalizedDescription=CoreSimulator is out of date. Current version (944.4.0) is older than build version (947.13.0).}). Simulator device support disabled.
ComputeTargetDependencyGraph
note: Building targets in dependency order
note: Target dependency graph (25 targets)
    Target 'deps' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_PNG' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_EXPAT' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_Boost' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_Cereal' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_Qhull' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_GLFW' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_OpenCSG' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_TBB' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_OpenVDB' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_CGAL' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_NLopt' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_OpenSSL' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_CURL' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_wxWidgets' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_OCCT' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_FREETYPE' in project 'OrcaSlicer-deps'
    Target 'dep_OCCT' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_FREETYPE' in project 'OrcaSlicer-deps'
    Target 'dep_FREETYPE' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
    Target 'dep_wxWidgets' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_PNG' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_EXPAT' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_JPEG' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_TIFF' in project 'OrcaSlicer-deps'
    Target 'dep_TIFF' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_PNG' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_JPEG' in project 'OrcaSlicer-deps'
    Target 'dep_JPEG' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
    Target 'dep_CURL' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_OpenSSL' in project 'OrcaSlicer-deps'
    Target 'dep_OpenSSL' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
    Target 'dep_NLopt' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
    Target 'dep_CGAL' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_Boost' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_GMP' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_MPFR' in project 'OrcaSlicer-deps'
    Target 'dep_MPFR' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_GMP' in project 'OrcaSlicer-deps'
    Target 'dep_GMP' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
    Target 'dep_OpenVDB' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_Boost' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_TBB' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_Blosc' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_OpenEXR' in project 'OrcaSlicer-deps'
    Target 'dep_OpenEXR' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
    Target 'dep_Blosc' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
    Target 'dep_TBB' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
    Target 'dep_OpenCSG' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'dep_GLEW' in project 'OrcaSlicer-deps'
    Target 'dep_GLEW' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
    Target 'dep_GLFW' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
    Target 'dep_Qhull' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
    Target 'dep_Cereal' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
    Target 'dep_Boost' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
    Target 'dep_EXPAT' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
    Target 'dep_PNG' in project 'OrcaSlicer-deps'
        ➜ Explicit dependency on target 'ZERO_CHECK' in project 'OrcaSlicer-deps'
    Target 'ZERO_CHECK' in project 'OrcaSlicer-deps' (no dependencies)

GatherProvisioningInputs

CreateBuildDescription

ExecuteExternalTool /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -v -E -dM -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk -x c -c /dev/null

Build description signature: fd463bc87bd77177a31f8bae10de98dd
Build description path: /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/XCBuildData/fd463bc87bd77177a31f8bae10de98dd.xcbuilddata
note: Run script build phase 'Generate CMakeFiles/ZERO_CHECK' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'ZERO_CHECK' from project 'OrcaSlicer-deps')
PhaseScriptExecution Generate\ CMakeFiles/ZERO_CHECK /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/ZERO_CHECK.build/Script-0499D897E14E3A4EEE5691E3.sh (in target 'ZERO_CHECK' from project 'OrcaSlicer-deps')
    cd /Users/lindner/VSCode/OrcaSlicer/deps
    /bin/sh -c /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/ZERO_CHECK.build/Script-0499D897E14E3A4EEE5691E3.sh
make: `/Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/CMakeFiles/cmake.check_cache' is up to date.

note: Run script build phase 'Generate CMakeFiles/dep_TBB' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_TBB' from project 'OrcaSlicer-deps')
PhaseScriptExecution Generate\ dep_TBB-prefix/src/dep_TBB-stamp/Debug/dep_TBB-configure /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_TBB.build/Script-0CA5679DFC492C075E583AC6.sh (in target 'dep_TBB' from project 'OrcaSlicer-deps')
    cd /Users/lindner/VSCode/OrcaSlicer/deps
    /bin/sh -c /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_TBB.build/Script-0CA5679DFC492C075E583AC6.sh
CMake Deprecation Warning at CMakeLists.txt:15 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.



PhaseScriptExecution Generate\ dep_Qhull-prefix/src/dep_Qhull-stamp/Debug/dep_Qhull-configure /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_Qhull.build/Script-6106A3A573E88338588AAA5A.sh (in target 'dep_Qhull' from project 'OrcaSlicer-deps')
    cd /Users/lindner/VSCode/OrcaSlicer/deps
    /bin/sh -c /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_Qhull.build/Script-6106A3A573E88338588AAA5A.sh
CMake Warning (dev) at CMakeLists.txt:67 (project):
  cmake_minimum_required() should be called prior to this top-level project()
  call.  Please see the cmake-commands(7) manual for usage documentation of
  both commands.
This warning is for project developers.  Use -Wno-dev to suppress it.


PhaseScriptExecution Generate\ dep_PNG-prefix/src/dep_PNG-stamp/Debug/dep_PNG-configure /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_PNG.build/Script-68BBC461202C410190483877.sh (in target 'dep_PNG' from project 'OrcaSlicer-deps')
    cd /Users/lindner/VSCode/OrcaSlicer/deps
    /bin/sh -c /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_PNG.build/Script-68BBC461202C410190483877.sh
CMake Deprecation Warning at CMakeLists.txt:14 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at CMakeLists.txt:15 (cmake_policy):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Symbol prefix: 
-- Configuring done (1.6s)
-- Generating done (0.0s)
CMake Error in CMakeLists.txt:
  The custom command generating

    /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/dep_PNG-prefix/src/dep_PNG-build/pnglibconf.c

  is attached to multiple targets:

    genvers
    gensym
    genfiles

  but none of these is a common dependency of the other(s).  This is not
  allowed by the Xcode "new build system".


CMake Generate step failed.  Build files cannot be regenerated correctly.
Command PhaseScriptExecution failed with a nonzero exit code

PhaseScriptExecution Generate\ dep_OpenSSL-prefix/src/dep_OpenSSL-stamp/Debug/dep_OpenSSL-configure /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_OpenSSL.build/Script-1AD3269EBBB8BFF5EDDDE833.sh (in target 'dep_OpenSSL' from project 'OrcaSlicer-deps')
    cd /Users/lindner/VSCode/OrcaSlicer/deps
    /bin/sh -c /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_OpenSSL.build/Script-1AD3269EBBB8BFF5EDDDE833.sh
Configuring OpenSSL version 1.1.1w (0x1010117fL) for darwin64-arm64-cc
Using os-specific seed configuration
Creating configdata.pm
Creating Makefile

**********************************************************************
***                                                                ***
***   OpenSSL has been successfully configured                     ***
***                                                                ***
***   If you encounter a problem while building, please open an    ***
***   issue on GitHub <https://github.com/openssl/openssl/issues>  ***
***   and include the output from the following command:           ***
***                                                                ***
***       perl configdata.pm --dump                                ***
***                                                                ***
***   (If you are new to OpenSSL, you might want to consult the    ***
***   'Troubleshooting' section in the INSTALL file first)         ***
***                                                                ***
**********************************************************************

PhaseScriptExecution Generate\ dep_NLopt-prefix/src/dep_NLopt-stamp/Debug/dep_NLopt-configure /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_NLopt.build/Script-2159BEBBEAA71F948E612AA2.sh (in target 'dep_NLopt' from project 'OrcaSlicer-deps')
    cd /Users/lindner/VSCode/OrcaSlicer/deps
    /bin/sh -c /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_NLopt.build/Script-2159BEBBEAA71F948E612AA2.sh
CMake Deprecation Warning at CMakeLists.txt:15 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.



PhaseScriptExecution Generate\ dep_JPEG-prefix/src/dep_JPEG-stamp/Debug/dep_JPEG-configure /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_JPEG.build/Script-9F6CA92C876C07FD481717F7.sh (in target 'dep_JPEG' from project 'OrcaSlicer-deps')
    cd /Users/lindner/VSCode/OrcaSlicer/deps
    /bin/sh -c /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_JPEG.build/Script-9F6CA92C876C07FD481717F7.sh
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.



PhaseScriptExecution Generate\ dep_GLFW-prefix/src/dep_GLFW-stamp/Debug/dep_GLFW-configure /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_GLFW.build/Script-FA2B14A8F774544A4516A0A3.sh (in target 'dep_GLFW' from project 'OrcaSlicer-deps')
    cd /Users/lindner/VSCode/OrcaSlicer/deps
    /bin/sh -c /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_GLFW.build/Script-FA2B14A8F774544A4516A0A3.sh

PhaseScriptExecution Generate\ dep_FREETYPE-prefix/src/dep_FREETYPE-stamp/Debug/dep_FREETYPE-download /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_FREETYPE.build/Script-630294091388231DFEE8BF93.sh (in target 'dep_FREETYPE' from project 'OrcaSlicer-deps')
    cd /Users/lindner/VSCode/OrcaSlicer/deps
    /bin/sh -c /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_FREETYPE.build/Script-630294091388231DFEE8BF93.sh
-- verifying file...
       file='/Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/../DL_CACHE/FREETYPE/freetype-2.12.1.tar.gz'
-- SHA256 hash of
    /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/../DL_CACHE/FREETYPE/freetype-2.12.1.tar.gz
  does not match expected value
    expected: 'efe71fd4b8246f1b0b1b9bfca13cfff1c9ad85930340c27df469733bbb620938'
      actual: 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'
-- File already exists but hash mismatch. Removing...
-- Downloading...
   dst='/Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/../DL_CACHE/FREETYPE/freetype-2.12.1.tar.gz'
   timeout='none'
   inactivity timeout='none'
-- Using src='https://mirror.ossplanet.net/nongnu/freetype/freetype-2.12.1.tar.gz'

PhaseScriptExecution Generate\ dep_GLEW-prefix/src/dep_GLEW-stamp/Debug/dep_GLEW-configure /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_GLEW.build/Script-9B94C2BF4A64E94D61EFC312.sh (in target 'dep_GLEW' from project 'OrcaSlicer-deps')
    cd /Users/lindner/VSCode/OrcaSlicer/deps
    /bin/sh -c /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_GLEW.build/Script-9B94C2BF4A64E94D61EFC312.sh
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.



PhaseScriptExecution Generate\ dep_Boost-prefix/src/dep_Boost-stamp/Debug/dep_Boost-download /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_Boost.build/Script-D785B0805432B6364CF71853.sh (in target 'dep_Boost' from project 'OrcaSlicer-deps')
    cd /Users/lindner/VSCode/OrcaSlicer/deps
    /bin/sh -c /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_Boost.build/Script-D785B0805432B6364CF71853.sh
-- verifying file...
       file='/Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/../DL_CACHE/Boost/boost-1.84.0.tar.gz'
-- SHA256 hash of
    /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/../DL_CACHE/Boost/boost-1.84.0.tar.gz
  does not match expected value
    expected: '4d27e9efed0f6f152dc28db6430b9d3dfb40c0345da7342eaa5a987dde57bd95'
      actual: '1bd9452a300a51beebe1061ec923d5aeba623c8f59ff6e2cf9100aea51a211cb'
-- File already exists but hash mismatch. Removing...
-- Downloading...
   dst='/Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/../DL_CACHE/Boost/boost-1.84.0.tar.gz'
   timeout='none'
   inactivity timeout='none'
-- Using src='https://github.com/boostorg/boost/releases/download/boost-1.84.0/boost-1.84.0.tar.gz'

WriteAuxiliaryFile /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_OpenEXR.build/Script-C0B255EEA437ADD29DC80AE5.sh (in target 'dep_OpenEXR' from project 'OrcaSlicer-deps')
    cd /Users/lindner/VSCode/OrcaSlicer/deps
    write-file /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_OpenEXR.build/Script-C0B255EEA437ADD29DC80AE5.sh

PhaseScriptExecution Generate\ dep_OpenSSL-prefix/src/dep_OpenSSL-stamp/Debug/dep_OpenSSL-build /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_OpenSSL.build/Script-EBC77FD3C801C03EB17574C5.sh (in target 'dep_OpenSSL' from project 'OrcaSlicer-deps')
    cd /Users/lindner/VSCode/OrcaSlicer/deps
    /bin/sh -c /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_OpenSSL.build/Script-EBC77FD3C801C03EB17574C5.sh
perl "-I." -Mconfigdata "util/dofile.pl" \
	    "-oMakefile" include/crypto/bn_conf.h.in > include/crypto/bn_conf.h
perl "-I." -Mconfigdata "util/dofile.pl" \
	    "-oMakefile" include/crypto/dso_conf.h.in > include/crypto/dso_conf.h
perl "-I." -Mconfigdata "util/dofile.pl" \
	    "-oMakefile" include/openssl/opensslconf.h.in > include/openssl/opensslconf.h
make depend && make _all
make: *** [all] Interrupt: 2

PhaseScriptExecution Generate\ dep_OpenEXR-prefix/src/dep_OpenEXR-stamp/Debug/dep_OpenEXR-configure /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_OpenEXR.build/Script-C0B255EEA437ADD29DC80AE5.sh (in target 'dep_OpenEXR' from project 'OrcaSlicer-deps')
    cd /Users/lindner/VSCode/OrcaSlicer/deps
    /bin/sh -c /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_OpenEXR.build/Script-C0B255EEA437ADD29DC80AE5.sh

note: Run script build phase 'Generate CMakeFiles/dep_CGAL' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_CGAL' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_Blosc' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_Blosc' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_OpenVDB' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_OpenVDB' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_OCCT' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_OCCT' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_CURL' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_CURL' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_OpenCSG' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_OpenCSG' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate dep_wxWidgets-prefix/src/dep_wxWidgets-stamp/Debug/dep_wxWidgets-update' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_wxWidgets' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate dep_wxWidgets-prefix/src/dep_wxWidgets-stamp/Debug/dep_wxWidgets-patch' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_wxWidgets' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/Debug/dep_wxWidgets-complete' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_wxWidgets' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_wxWidgets' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_wxWidgets' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_GMP' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_GMP' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_TIFF' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_TIFF' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_EXPAT' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_EXPAT' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_MPFR' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_MPFR' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_Cereal' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_Cereal' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/deps' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'deps' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_Qhull' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_Qhull' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_PNG' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_PNG' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_OpenSSL' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_OpenSSL' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_NLopt' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_NLopt' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_JPEG' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_JPEG' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_GLFW' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_GLFW' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_FREETYPE' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_FREETYPE' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_GLEW' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_GLEW' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_Boost' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_Boost' from project 'OrcaSlicer-deps')
note: Run script build phase 'Generate CMakeFiles/dep_OpenEXR' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'dep_OpenEXR' from project 'OrcaSlicer-deps')
** BUILD FAILED **


The following build commands failed:
	PhaseScriptExecution Generate\ dep_PNG-prefix/src/dep_PNG-stamp/Debug/dep_PNG-configure /Users/lindner/VSCode/OrcaSlicer/deps/build_arm64/build/OrcaSlicer-deps.build/Release/dep_PNG.build/Script-68BBC461202C410190483877.sh (in target 'dep_PNG' from project 'OrcaSlicer-deps')
(1 failure)
(base)lindner@M1pro OrcaSlicer %

nevack commented:
> Hi all
> 
> 
> 
> seems like the pr broke the script when running on arm64 local build. Anyone else encountered this? 

Hi, sorry for the inconvenience.
I will look into this in an hour.

It's very unusual as I also build locally on arm64 Mac.

Do you also have a build log with a failure?
Do you build with Xcode or Ninja generator?
Have you cleaned build dir to remove stale CMakr caches and configuration data?

This will help me a lot to triage.

igiannakas commented:
Same error above. Used to build with Xcode. Deleted the complete GitHub folder and reinstalled dependencies plus tried a full build but no change.

igiannakas commented:
As in running the basic script with no parameters.
